devito/data/utils.py

```diff
@@ -0,0 +1,259 @@
+import numpy as np
+
+from devito.tools import Tag, as_tuple, is_integer
+
+__all__ = ['Index', 'NONLOCAL', 'PROJECTED', 'index_is_basic', 'index_apply_modulo',
+           'index_dist_to_repl', 'convert_index', 'index_handle_oob',
+           'local_data_idx', 'mpi_index_maps']
+
+
+class Index(Tag):
+    pass
+NONLOCAL = Index('nonlocal')  # noqa
+PROJECTED = Index('projected')
```

Comment:
I wonder whether we can use this class for PARALLEL and SERIAL too ?

to be fair, I'm not sure I remember how I was using it, so I might be confusing things and talking BS...

Reply:
Hmm, would be nice but I don't think we can. Will have a look though.